src/Temporalio/Bridge/src/client.rs

```diff
@@ -0,0 +1,453 @@
+        "AddSearchAttributes" => rpc_call!(client, call, add_search_attributes),
+        "DeleteNamespace" => rpc_call!(client, call, delete_namespace),
+        "DeleteWorkflowExecution" => rpc_call!(client, call, delete_workflow_execution),
+        "ListClusters" => rpc_call!(client, call, list_clusters),
+        "ListSearchAttributes" => rpc_call!(client, call, list_search_attributes),
+        "RemoveRemoteCluster" => rpc_call!(client, call, remove_remote_cluster),
+        "RemoveSearchAttributes" => rpc_call!(client, call, remove_search_attributes),
+        rpc => Err(anyhow::anyhow!("Unknown RPC call {}", rpc)),
+    }
+}
+
+async fn call_test_service(client: &CoreClient, call: &RpcCallOptions) -> anyhow::Result<Vec<u8>> {
+    let rpc = call.rpc.to_str();
+    let mut client = client.clone();
+    match rpc {
+        "GetCurrentTime" => rpc_call!(client, call, get_current_time),
+        "LockTimeSkipping" => rpc_call!(client, call, lock_time_skipping),
+        "SleepUntil" => rpc_call!(client, call, sleep_until),
+        "Sleep" => rpc_call!(client, call, sleep),
+        "UnlockTimeSkippingWithSleep" => rpc_call!(client, call, unlock_time_skipping_with_sleep),
+        "UnlockTimeSkipping" => rpc_call!(client, call, unlock_time_skipping),
+        rpc => Err(anyhow::anyhow!("Unknown RPC call {}", rpc)),
+    }
+}
+
+async fn call_health_service(
+    client: &CoreClient,
+    call: &RpcCallOptions,
+) -> anyhow::Result<Vec<u8>> {
+    let rpc = call.rpc.to_str();
+    let mut client = client.clone();
+    match rpc {
+        "Check" => rpc_call!(client, call, check),
+        rpc => Err(anyhow::anyhow!("Unknown RPC call {}", rpc)),
+    }
+}
+
+fn rpc_req<P: prost::Message + Default>(
+    call: &RpcCallOptions,
+) -> anyhow::Result<tonic::Request<P>> {
+    let proto = P::decode(call.req.to_slice())?;
+    let mut req = tonic::Request::new(proto);
+    if call.metadata.size > 0 {
+        for (k, v) in call.metadata.to_str_map_on_newlines() {
+            req.metadata_mut()
+                .insert(MetadataKey::from_str(k)?, v.parse()?);
+        }
+    }
+    if call.timeout_millis > 0 {
+        req.set_timeout(Duration::from_millis(call.timeout_millis.into()));
+    }
+    Ok(req)
+}
+
+fn rpc_resp<P>(res: Result<tonic::Response<P>, tonic::Status>) -> anyhow::Result<Vec<u8>>
+where
+    P: prost::Message,
+    P: Default,
+{
+    Ok(res?.get_ref().encode_to_vec())
+    // TODO(cretz): Better error encoding
+    // match res {
+    //     Ok(resp) => Ok(resp.get_ref().encode_to_vec()),
+    //     Err(err) => {
+    //         Err(Python::with_gil(move |py| {
+    //             // Create tuple of "status", "message", and optional "details"
+    //             let code = err.code() as u32;
+    //             let message = err.message().to_owned();
+    //             let details = err.details().into_py(py);
+    //             RPCError::new_err((code, message, details))
+    //         }))
+    //     }
+    // }
```

Comment:
Commented out from Python, but the goal is to get proper advanced errors over to .NET (gRPC status details is something no other .NET gRPC user has :-))